### DISCUSSION TITLE
New Feature: Quick Actions

### DISCUSSION THREAD

**Discussion by jackMort** [jackMort/ChatGPT.nvim#39] [Announcements]
I am working on a  new feature that allows you to apply actions to a selected fragment of your text/code.

Some examples of how you can use this feature include:

- translating text
- grammar correction
- adding docstring
- ...

Here's a preview of the new feature in action:

[![Video preview](https://img.youtube.com/vi/EJRof1DwYbw/0.jpg)](https://www.youtube.com/watch?v=EJRof1DwYbw)

Do you have any other examples that could work for such actions, please add your ideas in comments

**Comment by ben-wall** [comment]
This is brilliant!
Would be good if we could have also the ability to make a custom request like this so that we can map it our own binding;
`<leader>air :ChatGPTQuickAction "reverse this text"<cr>`

**Comment by pitkling** [comment]
I was playing around with this, it's a pretty great feature, thanks!

I saw that a while ago you started to implement customizable quick actions. Is that feature already usable? I tried to define my own simple actions in a `.json` file and added the corresponding file path to the configuration option `paths_actions`, but it seems my custom actions are not picked up.

**Reply by pitkling** [reply]
Sorry, my bad. I misunderstood the user interface, now it works :).
